script.reddit.reader/resources/lib/slideshow.py

```diff
@@ -0,0 +1,1451 @@
+            
+            log( "queue size:%d" %q.qsize() )
+            #texts=[item[0] for item in q.queue]
+            #for i in images: self.log('   image: %s' %i)
+            #self.log('    %d images' % len(images))
+
+        return images
+
+
+    #for movie, audio or tv shows
+    def _get_json_images(self, method, key, prop):
+        self.log('_get_json_images start')
+        query = {
+            'jsonrpc': '2.0',
+            'id': 0,
+            'method': method,
+            'params': {
+                'properties': [prop],
+            }
+        }
+        response = json.loads(xbmc.executeJSONRPC(json.dumps(query)))
+        images = [
+            element[prop] for element
+            in response.get('result', {}).get(key, [])
+            if element.get(prop)
+        ]
+        self.log('_get_json_images end')
+        return images
+
+    def _get_folder_images(self, path):
+        self.log('_get_folder_images started with path: %s' % repr(path))
+        dirs, files = xbmcvfs.listdir(path)
+        images = [
+            xbmc.validatePath(path + f) for f in files
+            if f.lower()[-3:] in ('jpg', 'png')
+        ]
+        #if addon.getSetting('recursive') == 'true':
+        #    for directory in dirs:
+        #        if directory.startswith('.'):
+        #            continue
+        #        images.extend(
+        #            self._get_folder_images(
+        #                xbmc.validatePath('/'.join((path, directory, '')))
+        #            )
+        #        )
+        self.log('_get_folder_images ends')
+        return images
+
+    def hide_loading_indicator(self):
+        bg_img = xbmc.validatePath('/'.join(( ADDON_PATH, 'resources', 'skins', 'Default', 'media', self.BACKGROUND_IMAGE )))
+        #bg_img = self.BACKGROUND_IMAGE
+        self.loading_control.setAnimations([(
+            'conditional',
+            'effect=fade start=100 end=0 time=500 condition=true'
+        )])
+        self.background_control.setAnimations([(
+            'conditional',
+            'effect=fade start=0 end=100 time=500 delay=500 condition=true'
+        )])
+        self.background_control.setImage(bg_img)
+
+    def process_image(self, image_control, image_url):
+        # Needs to be implemented in sub class
+        raise NotImplementedError
+
+    def preload_image(self, image_url):
+        # set the next image to an unvisible image-control for caching
+        #self.log('preloading image: %s' % repr(image_url))
+        self.preload_control.setImage(image_url)
+        #self.log('preloading done')
+
+    def wait(self):
+        # wait in chunks of 500ms to react earlier on exit request
+        chunk_wait_time = int(CHUNK_WAIT_TIME)
+        remaining_wait_time = int(self.NEXT_IMAGE_TIME)
+        while remaining_wait_time > 0:
+            if self.exit_requested:
```

Comment:
There is an xbmc.Monitor that calls the function that sets this value when user presses any key